api/src/main/java/org/wildfly/transformer/TransformerBuilder.java

```diff
@@ -0,0 +1,155 @@
+        mappingWithDots = new HashMap<>();
+    }
+
+    /**
+     * Adds custom packages mapping defined in a configuration file.
+     * First this method tries to read the specified file from the file system.
+     * If file doesn't exist on the file system then class loader resources are inspected.
+     * The specified file <code>config</code> must exist either on the  file system or
+     * in class loader resources otherwise exception is thrown.
+     * Once this method is called it turns off <i>default packages mapping configuration</i>
+     * and <i>user provided configuration</i> is used instead.
+     *
+     * @param config packages mapping configuration file
+     * @return this builder instance
+     * @throws ConcurrentModificationException if this builder instance is used by multiple threads
+     * @throws IllegalStateException if either this or {@link #build()} method have been already called
+     * @throws IllegalArgumentException if method parameter is <code>null</code>
+     * or if method parameter equals to <code>empty string</code>
+     * or if file doesn't exist neither on the file system nor in class loader resources
+     * @throws IOException if some I/O error occurs while reading the configuration file <code>config</code>
+     */
+    public final TransformerBuilder setPackagesMapping(final String config) throws IOException {
+        // preconditions
+        if (thread != currentThread()) throw new ConcurrentModificationException("Builder instance used by multiple threads");
+        if (built) throw new IllegalStateException("Builder instance have been already closed");
+        if (mappingFile != null) throw new IllegalStateException("This method can be called only once");
+        if (config == null || "".equals(config)) throw new IllegalArgumentException("Parameter cannot be neither null nor empty string");
+        // implementation
+        final File userConfig = new File(config);
+        if (userConfig.exists() && userConfig.isFile()) {
+            mappingFile = new FileInputStream(userConfig);
+        } else {
+            mappingFile = TransformerBuilder.class.getResourceAsStream(config);
+        }
+        if (mappingFile == null) throw new IllegalArgumentException("Couldn't find specified config file neither on file system nor on class path");
+        return this;
+    }
+
+    /**
+     * Creates new resource transformer and closes this builder instance.
+     *
+     * @return new resource transformer
+     * @throws ConcurrentModificationException if this builder instance is used by multiple threads
+     * @throws IllegalStateException if this method have been already called or
+     * there was no packages mapping defined in configuration file or
+     * if packages mapping count in configuration file surpasses value <code>65535</code>
+     * @throws IllegalArgumentException if configuration file has invalid format or it contains identity package mapping
```

Comment:
Minor question, should "identity" be "identical"?

Reply:
With identity packing mapping I mean e.g. mapping: javax/servlet/=javax/servlet/
Is it proper english term? I am not native English speaker so I don't know for sure if should be "identity" or "identical". Can you help :)

Reply:
Thanks, that is what I thought you were trying to check for, "identical" would describe that better.

Identity is more like what something is, or who someone is.  For example, my passport proves who I am.

Reply:
Thanks @scottmarlow I fixed it plus I little bit enhanced the exception message to display detected identical package mapping to the API user.